x-pack/plugins/osquery/public/shared_components/osquery_results/index.tsx

```diff
@@ -23,41 +21,30 @@ import { OsqueryResult } from './osquery_result';
 const OsqueryActionResultsComponent: React.FC<OsqueryActionResultsProps> = ({
   agentIds,
   ruleName,
-  alertId,
+  actionItems,
   ecsData,
-}) => {
-  const { data: actionsData } = useAllLiveQueries({
-    filterQuery: { term: { alert_ids: alertId } },
-    activePage: 0,
-    limit: 100,
-    direction: Direction.desc,
-    sortField: '@timestamp',
-  });
-
-  return (
-    <div data-test-subj={'osquery-results'}>
-      {actionsData?.data.items.map((item, index) => {
-        const actionId = item.fields?.action_id?.[0];
-        const queryId = item.fields?.['queries.action_id']?.[0];
-        // const query = item.fields?.['queries.query']?.[0];
-        const startDate = item.fields?.['@timestamp'][0];
+}) => (
+  <div data-test-subj={'osquery-results'}>
+    {actionItems?.map((item, index) => {
+      const actionId = item.fields?.action_id?.[0];
+      const queryId = item.fields?.['queries.action_id']?.[0];
+      const startDate = item.fields?.['@timestamp'][0];
 
-        return (
-          <OsqueryResult
-            key={actionId + index}
-            actionId={actionId}
-            queryId={queryId}
-            startDate={startDate}
-            ruleName={ruleName}
-            agentIds={agentIds}
-            ecsData={ecsData}
-          />
-        );
-      })}
-      <EuiSpacer size="s" />
-    </div>
-  );
-};
+      return (
+        <OsqueryResult
+          key={actionId + index}
```

Comment:
please use something else than `index`, as it may cause rendering outdated data

Reply:
actionId should be enough 👍